server/src/main/java/io/druid/client/BrokerServerView.java

```diff
@@ -244,15 +244,15 @@ private void serverRemovedSegment(DruidServerMetadata server, DataSegment segmen
   {
 
     String segmentId = segment.getIdentifier();
-    final ServerSelector selector;
+    ServerSelector selector;
 
     synchronized (lock) {
       log.debug("Removing segment[%s] from server[%s].", segmentId, server);
 
       selector = selectors.get(segmentId);
       if (selector == null) {
         log.warn("Told to remove non-existant segment[%s]", segmentId);
```

Comment:
Maybe should rephrase the logging statement now since and change level to "info"?